geospaas_processing/downloaders.py

```diff
@@ -317,6 +318,11 @@ def download_dataset(self, dataset, download_directory):
                         f"Could not write the dowloaded file to {error.filename}") from error
                 else:
                     if downloaded:
+                        if self.store_address:
+                            dataset.dataseturi_set.update_or_create(
+                                service='DOWNLOAD_DIR', dataset=dataset,
+                                defaults={"uri": os.path.join(download_directory, file_name)}
+                                )
```

Comment:
```suggestion
                        if self.save_path:
                            dataset.dataseturi_set.update_or_create(
                                dataset=dataset,
                                uri = os.path.join(download_directory, file_name),
                                )
```

Reply:
@aperrin66 
@akorosov 
If we use `get_or_create` then we may have several addresses stored in the database that happens for the cases of changing the downloading directory when we run the downloading cli. 

What is the most valid address for copying the file? obviously the last successful attempt of download. So copying must be down with the source of last downloaded ones. Old addresses might be removed!

Otherwise, if we use `update_or_create` there is always one record in the dataset_uri model which points to the most recent location of the file. This way we can use `get` of django and not concern about the duplicating addresses in the database. 

The way I wrote the `update_or_create` is the way that it searches based on the service='DOWNLOAD_DIR' and then it updates the address if necessary. So, "uri" must come as a default for `update_or_create` in order to be updated if necessary: https://docs.djangoproject.com/en/3.1/ref/models/querysets/#update-or-create

This suggestion does not doing that algorithm.

Reply:
Maybe this is not the place to check for the validity of existing DatasetURIs. We talked about having a separate checking mechanism, maybe a script executed regularly which would remove the stale DatasetURIs.
Considering this, maybe you can just add a new DatasetURI here and not bother with the existing ones.

Reply:
DOne in the next commit.